pkg/queue/concurrency_state.go

```diff
@@ -19,30 +19,45 @@ package queue
 import (
 	"bytes"
 	"fmt"
+	"io/ioutil"
 	"net/http"
 	"os"
 	"sync"
+	"time"
 
 	"go.uber.org/atomic"
 	"go.uber.org/zap"
 )
 
 //nolint:gosec // Filepath, not hardcoded credentials
 const ConcurrencyStateTokenVolumeMountPath = "/var/run/secrets/tokens"
+const ConcurrencyStateTokenName = "state-token"
+const ConcurrencyStateToken = ConcurrencyStateTokenVolumeMountPath + "/" + ConcurrencyStateTokenName
```

Comment:
these consts having to be exported and shared between pkg/deploy and pkg/queue are a bit sad, wonder if there's a way we could avoid that. What if we got the token path from an env variable? That seems a bit more decoupled and maybe a bit cleaner (this is how we used to handle a similar thing in the past where we [looked for some info via the downward api](https://github.com/knative/serving/blob/f1582404be275d6eaaf89ccd908fb44aef9e48b5/cmd/queue/main.go#L112)).

Reply:
bump ^

Reply:
FWIW, exporting/sharing constant names seems to be a fairly common pattern to pass values to pkg/deploy (queue.Name and queue.RequestQueueDrainPath are passed this way, as well as several from networking). Would it make sense to convert them _all_ to environmental variables (all meaning the QP ones)? If so, perhaps that would be better done in a separate PR.

If that's the route we want to go, for now I could either leave the CE constants where they are or move them to pkg/queue/constants.go

Reply:
I'm not really convinced queue.Name and queue.RequestQueueDrainPath (or the port related constants in networking) are a very good parallel here. The main issue is that we're exporting three separate related constants, which makes the api of pkg/queue more complicated (and coupled) than it needs to be. If instead we passed the mounted location of the token, this would be a fair bit simpler and cleaner.

FWIW this is what we did both times we previously did a feature involving mounting a volume in QP ([1](https://github.com/knative/serving/blob/cfa8e8764fc15e6b81d20d0069787947ee03fdf9/pkg/reconciler/revision/resources/queue.go#L333-L334), [2](https://github.com/knative/serving/blob/cfa8e8764fc15e6b81d20d0069787947ee03fdf9/pkg/reconciler/revision/resources/queue.go#L327-L328)), and it seemed quite nice that way.

Reply:
Fair enough, was just checking if we should also update the other ones.

Reply:
done